src/laser.c

```diff
@@ -105,15 +105,20 @@ void laser_print_entry(struct laser_dirent *entry, const char *color,
 }
 
 static laser_color_type laser_color_for_format(const char *filename)
-{
-    if (laser_checktype(filename, laser_archiveformats))
-        return LASER_COLOR_ARCHIVE;
-    else if (laser_checktype(filename, laser_mediaformats))
-        return LASER_COLOR_MEDIA;
-    else if (laser_checktype(filename, laser_documentformats))
-        return LASER_COLOR_DOCUMENT;
-
-    return LASER_COLOR_FILE;
+
+{ //create fd because we use lseek to return on Bof position each laser_checktype_magic call inside
+    // and we can keep fd while checking types, even if fd is not valid
+    int fd = open(filename, O_RDONLY);
```

Comment:
I don't see how this reduces the amount of fd open call, rather dosen't this increase the open calls? In the main branch we only call open if checking extension fails. This calls open before even checking on the extension.